File: <100/067/067.py
def main() -> None:
    filename = "067.txt"
    F: list[list] = []
    with open(filename, 'r') as f:
        for row in f:
            if row[-1] == '\n':
                row = row[:-1]
            r = [int(c) for c in row.split(' ')]
            F.append(r)
    H = len(F)

    dp = [[0 for _ in range(H)] for __ in range(H)]
    dp[0][0] = F[0][0]
    for r in range(H-1):
        for c in range(r+1):
            dp[r+1][c] = max(dp[r+1][c], dp[r][c] + F[r+1][c])
            dp[r+1][c+1] = max(dp[r+1][c+1], dp[r][c] + F[r+1][c+1])

    answer = max(dp[-1])
    print(answer)
    return


if __name__ == '__main__':
    main()

File: <100/067/067.cpp
#include<iostream>
#include<fstream>
using namespace std;
#define MAX 100

int H = MAX, answer, dp[MAX+5][MAX+5], F[MAX+5][MAX+5];
string filename = "067.txt";

int main(){
  ifstream in(filename);
  cin.rdbuf(in.rdbuf());

  for(int h = 0; h < H; ++h){
    for(int c = 0; c <= h; ++c){
      cin >> F[h][c];
    }
  }

  dp[0][0] = F[0][0];
  for(int h = 0; h < H-1; ++h){
    for(int c = 0; c <= h; ++c){
      for(int d = 0; d <= 1; ++d){
        int c_ = c + d;
        if(h+1 < c_) continue;
        dp[h+1][c_] = max(dp[h+1][c_], dp[h][c] + F[h+1][c_]);
      }
    }
  }

  for(int c = 0; c < H; ++c) answer = max(answer, dp[H-1][c]);

  cout << answer << endl;
  return 0;
}
